File: Cpp/Move Zeroes.cc
# 17ms, 32.85%
class Solution {
public:
    void moveZeroes(vector<int> &nums) {
        int ins_index = 0, end = nums.size();
        while (nums[ins_index] != 0) ++ins_index;
        int trav = ins_index + 1;
        while (trav < end) {
            if (nums[trav] != 0)
                swap(nums[trav], nums[ins_index++]);
            ++trav;
        }
    }
};

File: Java/Move Zeroes.java
// 0ms, 100.00%; 39.7MB, 14.69%
class Solution {
    public static void moveZeroes(int[] nums) {
        int i = 0;
        for (int j = 1; j < nums.length; ++j) {
            if (nums[i] != 0) {
                ++i;
            } else if (nums[j] != 0) {
                nums[i++] = nums[j];
                nums[j] = 0;
            }
        }
    }
}
